orca-clouddriver/src/main/groovy/com/netflix/spinnaker/orca/clouddriver/pipeline/servergroup/rollback/ExplicitRollback.groovy

```diff
@@ -124,10 +127,73 @@ class ExplicitRollback implements Rollback {
     }
 
     stages << disableServerGroupStage
-    stages << buildApplySourceServerGroupCapacityStage(parentStage, parentStage.mapTo(ResizeStrategy.Source))
+
+    // only restore the min if it was pinned, i.e. there was a resize
+    if (resizeStage != null) {
+      stages << buildApplySourceServerGroupCapacityStage(parentStage, parentStage.mapTo(ResizeStrategy.Source))
+    }
     return stages
   }
 
+  @Nullable TargetServerGroup lookupServerGroup(Stage parentStage, String serverGroupName) {
+    def fromContext = parentStage.mapTo(ResizeStrategy.Source)
+
+    try {
+      // TODO: throw some retries in there?
+      return oortHelper.getTargetServerGroup(
+        fromContext.credentials,
+        serverGroupName,
+        fromContext.location
+      ).orElse(null)
+    } catch(Exception e) {
+      log.error('Skipping resize stage because there was an error looking up {}', serverGroupName, e)
+      return null
+    }
+  }
+
+  @Nullable Stage buildResizeStage(Stage parentStage) {
+    TargetServerGroup rollbackServerGroup = lookupServerGroup(parentStage, rollbackServerGroupName)
+    if (!rollbackServerGroup) {
+      return null
+    }
+
+    TargetServerGroup restoreServerGroup = lookupServerGroup(parentStage, restoreServerGroupName)
+    if (!restoreServerGroup) {
+      return null
+    }
+
+    // we don't want to scale down restoreServerGroupName if rollbackServerGroupName is smaller for some reason
+    ResizeStrategy.Capacity newRestoreCapacity = [
+      max: Math.max(rollbackServerGroup.capacity.max, restoreServerGroup.capacity.max),
+      desired: Math.max(rollbackServerGroup.capacity.desired, restoreServerGroup.capacity.desired)
+    ]
+
+    // let's directly produce a capacity with a pinned min instead of relying on the resize stage
+    newRestoreCapacity.min = newRestoreCapacity.desired
+
+    ResizeStrategy.Capacity currentCapacity = restoreServerGroup.capacity
+    if (currentCapacity == newRestoreCapacity) {
+      log.info('Skipping resize stage because the current capacity of the restore server group {} would be unchanged ({})',
+        restoreServerGroupName, newRestoreCapacity)
+      return null
+    }
+
+    Map resizeServerGroupContext = new HashMap(parentStage.context) + [
+      action                       : ResizeStrategy.ResizeAction.scale_exact.toString(),
+      capacity                     : newRestoreCapacity.asMap(),
+      asgName                      : restoreServerGroupName,
+      targetLocation               : restoreServerGroup.getLocation(),
+      account                      : restoreServerGroup.credentials,
+      cloudProvider                : restoreServerGroup.cloudProvider,
+      pinMinimumCapacity           : true,
```

Comment:
wy do you need this if you already set min=desired

Reply:
it's not needed, I thought I would just leave it like that to express the intent anyway